### PR TITLE
Added --dns-domain flag to localkube

### PR DESCRIPTION
**Pull request by aaron-prindle**
Ref: #1309

### CODE REVIEW

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -217,6 +219,7 @@ func init() {
 	startCmd.Flags().StringArrayVar(&dockerEnv, "docker-env", nil, "Environment variables to pass to the Docker daemon. (format: key=value)")
 	startCmd.Flags().StringArrayVar(&dockerOpt, "docker-opt", nil, "Specify arbitrary flags to pass to the Docker daemon. (format: key=value)")
 	startCmd.Flags().String(apiServerName, constants.APIServerName, "The apiserver name which is used in the generated certificate for localkube/kubernetes.  This can be used if you want to make the apiserver available from outside the machine")
+	startCmd.Flags().String(dnsDomain, "", "The cluster dns domain name used in the kubernetes cluster (in this case used by localkube)")
```
Comment (r2d4) [inline cmd/minikube/cmd/start.go]: nit: I think ` (in this case used by localkube)` is clear from the context

File: pkg/minikube/cluster/types.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type KubernetesConfig struct {
 	KubernetesVersion string
 	NodeIP            string
 	APIServerName     string
+	DNSDomain         string
```
Comment (r2d4) [inline pkg/minikube/cluster/types.go]: It doesnt actually look like this is used yet?

File: pkg/minikube/cluster/commands.go [hunk 3]
```diff
@@ -122,6 +122,10 @@ func GenLocalkubeStartCmd(kubernetesConfig KubernetesConfig) (string, error) {
 		flagVals = append(flagVals, "--apiserver-name="+kubernetesConfig.APIServerName)
 	}
 
+	if kubernetesConfig.DNSDomain != "" {
+		flagVals = append(flagVals, "--dns-domain="+kubernetesConfig.DNSDomain)
+	}
+
```
Comment (r2d4) [inline pkg/minikube/cluster/commands.go]: Is there a default value?  If so we have to be careful just like the --apiserver-name flag

Comment (aaron-prindle) [inline pkg/minikube/cluster/commands.go]: No there isn't a default value

File: cmd/minikube/cmd/start.go (hunk 1 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/start.go]: changed

File: pkg/minikube/cluster/types.go (hunk 2 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/types.go]: its passed to localkube in commands.go

Comment (r2d4) [inline pkg/minikube/cluster/types.go]: Ah I forgot that the issue was that the flag already existed but there was no way to configure it.

**Pull request merged**